mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,384 @@
+
+2. To make the ``EmailTagHelper`` class available to all our Razor views, we will add the ``addTagHelper`` directive to the *Views/_ViewImports.cshtml* file:  
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/Views/_ViewImports.cshtml
+   :language: aspx-cs
+   :lines: 1-3
+   :emphasize-lines: 3
+
+To add a tag helper to a view, you first add the fully qualified name (``AuthoringTagHelpers.TagHelpers.EmailTagHelper``), and then the DLL name (*AuthoringTagHelpers*). You typically add tag helpers to the *Views/_ViewImports.cshtml* file as we've done here, but it's more common to use wildcard syntax to bring in all tag helpers with one line, as the following shows:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/Views/_ViewImportsCopy.cshtml
+   :language: aspx-cs
+   :lines: 2-4
+   :emphasize-lines: 2,3
+
+When you use the wildcard syntax, you don't need to use the fully qualified name. Also, note that the second line brings in the `ASP.NET 5 MVC 6 tag helpers <https://github.com/aspnet/Mvc/tree/dev/src/Microsoft.AspNet.Mvc.TagHelpers>`__ using the wild card syntax (those helpers are discussed in :doc:`intro`.) It's the ``@addTagHelper`` command that makes the tag helper available to the Razor view.
+   
+3. Update the markup in the *Views/Home/Contact.cshtml* file with these changes:
+
+.. literalinclude::  authoring/sample/TagHlp/src/AuthoringTagHelpers/Views/Home/Contact.cshtml
+   :language: aspx-cs
+   :emphasize-lines: 15-16
+   :lines: 1-17
+   
+4. Run the app and use your favorite browser to view the HTML source so you can verify that the email tag is replaced with anchor markup (``<a>Support</a>``). Support is rendered as a link, but it doesn't have an ``href`` attribute to make it functional. We'll fix that in the next section.
+
+Note: Like `HTML tags and attributes <http://www.w3.org/TR/html-markup/documents.html#case-insensitivity>`__, tags, class names and attributes in Razor, and C# are not case-sensitive.
+
+A working email Tag Helper
+----------------------------------
+In this section, we will update the ``EmailTagHelper`` so that it will create a valid anchor tag for email. We'll update our tag helper to take information from a Razor view (in the form of a ``mail-to`` attribute) and use that in generating the anchor.
+
+Update the ``EmailTagHelper`` class with the following:
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/EmailTagHelperMailTo.cs
+   :lines: 5-24
+   :dedent: 3 
+
+Notes:
+
+- The ASP.NET 5 runtime translates C# Pascal-cased classes and property names for tag helpers into `lower kebab case <http://stackoverflow.com/questions/11273282/whats-the-name-for-dash-separated-case/12273101#12273101>`__. Therefore, to use the ``MailTo`` attribute, you'll use ``<email mail-to="value"/>``.
```

Comment:
> The ASP.NET 5 runtime translates C# Pascal-cased classes and property names for tag helpers into `lower kebab case <http://stackoverflow.com/questions/11273282/whats-the-name-for-dash-separated-case/12273101#12273101>`__

`Pascal-cased class and property names for tag helpers are translated into their ``lower kebab case <http://stackoverflow.com/questions/11273282/whats-the-name-for-dash-separated-case/12273101#12273101>``__ equivalent`